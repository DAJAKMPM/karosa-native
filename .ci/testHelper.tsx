import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
