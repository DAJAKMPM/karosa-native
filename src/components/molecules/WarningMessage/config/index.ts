/**
 *
 * @format
 *
 */

export const NUM_LINES = 4;
export const CLOSE_ICON = 14;

export const WARNING_ICON = {
  HEIGHT: 19,
  WIDTH: 22,
};
