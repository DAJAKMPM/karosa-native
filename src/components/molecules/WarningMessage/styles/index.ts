/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { DIMENS, SPACING, theme } from "@app/styles";

const WarningMessageStyles = StyleSheet.create({
  container: {
    backgroundColor: theme.colors.gold15,
    height: DIMENS.screenHeight * 0.2,
  },
  content: {
    flexDirection: "row",
  },
  warningIcon: {
    marginRight: SPACING.xss,
    marginTop: SPACING.xs,
  },
  closeIcon: {
    marginLeft: "auto",
  },
  textContainer: {
    alignItems: "center",
    justifyContent: "center",
    paddingLeft: SPACING.xl,
    paddingRight: SPACING.lg,
  },
  text: {
    ...theme.textRegular,
    fontWeight: "400",
    color: theme.colors.dark30,
    textAlign: "left",
  },
});

export default WarningMessageStyles;
