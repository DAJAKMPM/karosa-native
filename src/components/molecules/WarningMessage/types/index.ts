/**
 *
 * @format
 *
 */

export type PropsType = {
  message: string;
};
