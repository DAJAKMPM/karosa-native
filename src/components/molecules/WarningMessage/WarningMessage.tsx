/**
 *
 * WarningMessage
 * @format
 *
 */

import React, { FC, Fragment, useState } from "react";
import { TouchableWithoutFeedback, View } from "react-native";
import { ListItem } from "react-native-elements";
import Text from "@app/atoms/Text";
import Icon from "@app/atoms/Icon";

import type { PropsType } from "./types";
import { CLOSE_ICON, NUM_LINES, WARNING_ICON } from "./config";
import WarningMessageStyles from "./styles";

const WarningMessage: FC<PropsType> = (props) => {
  const { message } = props;

  const [hide, setHide] = useState(false);

  if (hide) return <Fragment />;

  return (
    <ListItem containerStyle={WarningMessageStyles.container}>
      <Icon
        group="products"
        name="warning"
        width={WARNING_ICON.WIDTH}
        height={WARNING_ICON.HEIGHT}
        extraStyle={WarningMessageStyles.warningIcon}
      />
      <ListItem.Content style={WarningMessageStyles.textContainer}>
        <Text
          numberOfLines={NUM_LINES}
          text={message}
          textStyle={WarningMessageStyles.text}
        />
      </ListItem.Content>
      <TouchableWithoutFeedback onPress={() => setHide(true)}>
        <View>
          <Icon
            group="common"
            name="close"
            width={CLOSE_ICON}
            height={CLOSE_ICON}
            extraStyle={WarningMessageStyles.closeIcon}
          />
        </View>
      </TouchableWithoutFeedback>
    </ListItem>
  );
};

export default WarningMessage;
