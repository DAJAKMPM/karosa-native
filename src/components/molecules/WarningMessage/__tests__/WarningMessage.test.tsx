import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import WarningMessage from "../WarningMessage";

describe("WarningMessage Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<WarningMessage message="test" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
