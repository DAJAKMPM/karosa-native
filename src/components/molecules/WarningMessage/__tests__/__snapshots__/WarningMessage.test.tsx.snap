// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningMessage Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#FFFACD",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "height": 186.76000000000002,
          "padding": 16,
        }
      }
      testID="padView"
    >
      <View>
        <SvgMock
          height={19}
          style={
            Object {
              "marginRight": 2,
              "marginTop": 2,
            }
          }
          width={22}
        />
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
        theme={
          Object {
            "colors": Object {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": Object {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <Text
          numberOfLines={4}
          style={
            Object {
              "color": "#4F4F4F",
              "fontFamily": "SFProText-Regular",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 18.75,
              "textAlign": "left",
            }
          }
        >
          test
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingLeft": 16,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View>
          <SvgMock
            height={14}
            style={
              Object {
                "marginLeft": "auto",
              }
            }
            width={14}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
